packages/web/src/pages/sign-up-page/utils/loginDetailsSchema.ts

```diff
@@ -0,0 +1,22 @@
+import { z } from 'zod'
+
+import { isNotCommonPassword } from 'utils/commonPasswordCheck'
+
+import { emailSchema } from './emailSchema'
+
+// Due to issue with zod merge, manually rewriting
+// https://github.com/colinhacks/zod/issues/454
+export const loginDetailsSchema = z
+  .object({
+    password: z
+      .string()
+      .regex(/\d/, { message: 'hasNumber' })
+      .min(8, { message: 'minLength' })
+      .refine(isNotCommonPassword, { message: 'notCommon' }),
+    confirmPassword: z.string()
```

Comment:
can we just merge `passwordSchema`?

Reply:
so annoyingly you cannot merge a refined schema, since .refine returns an "effect". there appears to be no clean workaround, apart from defining the .object and .effect as separate constants, and then interleaving them back in... here is the context: https://github.com/colinhacks/zod/issues/454